src/core/thread/link_metrics.hpp

```diff
@@ -71,16 +71,12 @@ namespace LinkMetrics {
  * @{
  */
 
-/**
- * This class implements Thread Link Metrics query and management.
- *
- */
-class LinkMetrics : public InstanceLocator, private NonCopyable
-{
-    friend class ot::Neighbor;
-    friend class ot::UnitTester;
+#if OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE
 
+class LinkMetricsInitiator : public InstanceLocator, private NonCopyable
```

Comment:
How about renaming from `LinkMetricsInitiator` -> `Initiator`?
Since we are already under `LinkMetrics` namespace (so the full name becomes `LinkMetrics::Initiator`).